contracts/AccountRulesList.sol

```diff
@@ -76,4 +101,36 @@ contract AccountRulesList {
             return (false, address(0));
         }
     }
+
+    function getAll() internal view returns (address[] memory) {
+        uint listSize = list.sizeOf();
+        if (listSize == 0) {
+            return new address[](0);
+        }
+
+        address[] memory allAddresses = new address[](listSize);
+
+        bool hasNext = true;
+        uint counter = 0;
+        uint pointer = 0;
+
+        while(hasNext) {
+            (bool nodeExists, uint256 prev, uint256 next) = list.getNode(pointer);
+            if (nodeExists) {
+                if (pointer > 0) {
+                    allAddresses[counter++] = address(pointer);
+                }
+
+                if (next != 0) {
+                    pointer = next;
+                } else {
+                    hasNext = false;
+                }
+            }
+            //Getting rid of unused variable warning
```

Comment:
did we not have a better way of doing this?

Reply:
Copied from AdminList.sol ...